subsys/logging/log_output.c

```diff
@@ -216,14 +278,25 @@ static int timestamp_print(const struct log_output *output,
 			length = print_formatted(output, "%s.%06uZ ",
 						 time_str, ms * 1000U + us);
 #else
+			struct YMD_date date;
+
+			get_YMD_from_seconds(total_seconds, &date);
+			hours = hours % 24;
 			length = print_formatted(output,
-					"1970-01-01T%02u:%02u:%02u.%06uZ ",
+					"%04u-%02u-%02uT%02u:%02u:%02u.%06uZ ",
+					date.year, date.month, date.day,
 					hours, mins, seconds, ms * 1000U + us);
 #endif
 		} else {
+#if !IS_ENABLED(CONFIG_LOG_OUTPUT_FORMAT_LINUX_TIMESTAMP)
```

Comment:
i would recommend using 
```
if (!IS_ENABLED(CONFIG_LOG_OUTPUT_FORMAT_LINUX_TIMESTAMP)) {
}
```
as it seems to me that there might be some warnings with unused variables (e.g. `mins`) in certain configurations.

Reply:
You're right: I fixed this one